hazelcast-hibernate5/src/main/java/com/hazelcast/hibernate/distributed/IMapRegionCache.java

```diff
@@ -156,16 +158,13 @@ public long size() {
         return map.size();
     }
 
+    /**
+     * Returns the total in-memory cost in bytes (including IMap, Near Cache, backup, Merkle trees)
+     * or -1 if using Hazelcast Client
+     */
     @Override
     public long getSizeInMemory() {
-        long size = 0;
-        for (final Object key : map.keySet()) {
-            final EntryView entry = map.getEntryView(key);
-            if (entry != null) {
-                size += entry.getCost();
-            }
-        }
-        return size;
+        return isMember ? map.getLocalMapStats().getHeapCost() : -1;
```

Comment:
```
LocalMapStats#getHeapCost()

When InMemoryFormat#OBJECT is used, the heapcost is zero.
```
So, is that metric `(getSizeInMemory)` ignored when object format is configured for cache regions?

Reply:
Hibernate won't ignore it, but users should.

Keep in mind that this behaviour is consistent with what was there before. I did some digging and found this:

https://github.com/hazelcast/hazelcast/blob/97ff57ba92dc1fbfa77720885e7db40757999080/hazelcast/src/main/java/com/hazelcast/map/impl/record/ObjectRecord.java#L31-L36